File: app/src/main/java/de/thu/recipebook/FavoritesDbHelper.java
package de.thu.recipebook;

import android.content.Context;
import android.database.sqlite.SQLiteDatabase;
import android.database.sqlite.SQLiteOpenHelper;
import android.provider.BaseColumns;

public class FavoritesDbHelper extends SQLiteOpenHelper {
    public static final int DATABASE_VERSION = 1;
    public static final String DATABASE_NAME = "Recipe.db";
    public static final String FAVORITES_TABLE = "favorites";

    private static final String SQL_CREATE_ENTRIES = "CREATE TABLE " + FAVORITES_TABLE + " ("
            + BaseColumns._ID + " STRING PRIMARY KEY)";

    private static final String SQL_DELETE_ENTRIES = "DROP TABLE IF EXISTS " + FAVORITES_TABLE;

    public FavoritesDbHelper(Context context) {
        super(context, DATABASE_NAME, null, DATABASE_VERSION);
    }

    @Override
    public void onCreate(SQLiteDatabase sqLiteDatabase) {
        sqLiteDatabase.execSQL(SQL_CREATE_ENTRIES);
    }

    @Override
    public void onUpgrade(SQLiteDatabase sqLiteDatabase, int i, int i1) {
        sqLiteDatabase.execSQL(SQL_DELETE_ENTRIES);
        onCreate(sqLiteDatabase);
    }
}

File: Exercise 9/Kotlin/src/main/java/de/thu/recipebook/databases/FavoritesDbHelper.kt
package de.thu.recipebook.databases

import android.content.Context
import android.database.sqlite.SQLiteDatabase
import android.database.sqlite.SQLiteOpenHelper
import android.provider.BaseColumns

class FavoritesDbHelper(context: Context?) :
    SQLiteOpenHelper(context, DATABASE_NAME, null, DATABASE_VERSION) {
    override fun onCreate(sqLiteDatabase: SQLiteDatabase) {
        sqLiteDatabase.execSQL(SQL_CREATE_ENTRIES)
    }

    override fun onUpgrade(sqLiteDatabase: SQLiteDatabase, i: Int, i1: Int) {
        sqLiteDatabase.execSQL(SQL_DELETE_ENTRIES)
        onCreate(sqLiteDatabase)
    }

    companion object {
        const val DATABASE_VERSION = 1
        const val DATABASE_NAME = "Recipe.db"
        const val FAVORITES_TABLE = "favorites"
        private const val SQL_CREATE_ENTRIES = ("CREATE TABLE " + FAVORITES_TABLE + " ("
                + BaseColumns._ID + " STRING PRIMARY KEY)")
        private const val SQL_DELETE_ENTRIES = "DROP TABLE IF EXISTS " + FAVORITES_TABLE
    }
}